pkg/daemon/daemon.go

```diff
@@ -471,35 +471,40 @@ func ReexecuteForTargetRoot(target string) error {
 	} else {
 		klog.Info("assuming we can use container binary chroot() to host")
 	}
-	sourceBinary := "/usr/bin/machine-config-daemon" + sourceBinarySuffix
-	src, err := os.Open(sourceBinary)
-	if err != nil {
-		return fmt.Errorf("opening %s: %w", sourceBinary, err)
-	}
-	defer src.Close()
 
 	targetBinBase := "run/bin/machine-config-daemon"
 	targetBin := filepath.Join(target, targetBinBase)
-	targetBinDir := filepath.Dir(targetBin)
-	if _, err := os.Stat(targetBinDir); err != nil {
-		if err := os.Mkdir(targetBinDir, 0o755); err != nil {
-			return fmt.Errorf("mkdir %s: %w", targetBinDir, err)
+
+	// Be idempotent
+	if _, err := os.Stat(targetBin); err != nil {
+		sourceBinary := "/usr/bin/machine-config-daemon" + sourceBinarySuffix
+		src, err := os.Open(sourceBinary)
+		if err != nil {
+			return fmt.Errorf("opening %s: %w", sourceBinary, err)
 		}
-	}
+		defer src.Close()
 
-	f, err := os.Create(targetBin)
-	if err != nil {
-		return fmt.Errorf("writing %s: %w", targetBin, err)
-	}
-	if _, err := io.Copy(f, src); err != nil {
+		targetBinDir := filepath.Dir(targetBin)
+		if _, err := os.Stat(targetBinDir); err != nil {
+			if err := os.Mkdir(targetBinDir, 0o755); err != nil {
```

Comment:
Are we making the assumption that the error of the above is errnotfound? Might be helpful to make it explicit

Reply:
umm, I am not sure adding here additional error check will add much value. Irrespective of what is the error, we are returning the error anyway with no further action needed.

Reply:
Sorry, I meant the error return from os.Stat(), must like my other comment below, not the os.Mkdir

Reply:
fixed